scipy/spatial/transform/_rotation.pyx

```diff
@@ -2806,54 +2833,237 @@ cdef class Rotation:
+        if N > 1 and (weights.shape[0] != N):
+            raise ValueError("Expected `weights` to have number of values "
+                             "equal to number of input vectors, got "
+                             "{} values and {} vectors.".format(
+                                 weights.shape[0], N))
+        if (weights < 0).any():
+            raise ValueError("`weights` may not contain negative values")
+
+        # For the special case of a single vector pair, we use the infinite
+        # weight code path
+        if N == 1:
+            weight_is_inf = np.array([True])
+        else:
+            weight_is_inf = np.isposinf(weights)
+        n_inf = np.sum(weight_is_inf)
+
+        # Check for an infinite weight, which indicates that the corresponding
+        # vector pair is the primary unmoving reference to which we align the
+        # other secondary vectors
+        if n_inf > 1:
+            raise ValueError("Only one infinite weight is allowed")
+
+        elif n_inf == 1:
+            if return_sensitivity:
+                raise ValueError("Cannot return sensitivity matrix with an "
+                                 "infinite weight or one vector pair")
+            a_pri = a[weight_is_inf]
+            b_pri = b[weight_is_inf]
+            a_pri /= _norm3(a_pri[0])
+            b_pri /= _norm3(b_pri[0])
+
+            # We first find the minimum angle rotation between the primary
+            # vectors.
+            cross = np.cross(b_pri[0], a_pri[0])
+            theta = atan2(_norm3(cross), np.dot(a_pri[0], b_pri[0]))
+            if theta < 1e-3:  # small angle Taylor series approximation
+                theta2 = theta * theta
+                r = cross * (1 + theta2 / 6 + theta2 * theta2 * 7 / 360)
+            else:
+                r = cross * theta / np.sin(theta)
+            R_pri = cls.from_rotvec(r)
+
+            if N == 1:
+                # While the else clause below will work for a single pair of
+                # vectors (~weight_is_inf is empty and phi will be 0), it's
+                # more efficient to skip those calculations and just return the
+                # primary rotation here
+                R_opt = R_pri
+            else:
+                a_non_inf = a[~weight_is_inf]
+                b_non_inf = b[~weight_is_inf]
+                weights_non_inf = weights[~weight_is_inf]
+
+                # We apply the first rotation to the b vectors to align the
+                # primary vectors, resulting in vectors c. The secondary
+                # vectors must now be rotated about that primary vector to best
+                # align c to a.
+                c_non_inf = R_pri.apply(b_non_inf)
+
+                # Calculate vector components for the angle calculation. The
+                # angle phi to rotate a single 2D vector C to align to 2D
+                # vector A in the xy plane can be found with the equation
+                # phi = atan2(cross(C, A), dot(C, A)).
```

Comment:
Yes, it's true. And if you have many vectors the same formula can be applied, but with sums over all vectors (possibly weighted). It can be derived rigorously (in 2D) by finding a stationary point of a function of one argument (the angle phi).

The trick in 3D is to consider projections of vectors into the plane. Let's denote the normal unit vector as `u` (in our case is `a_pri`). Let `Cp` and `Ap` be the orthogonal projections. Because they still live in 3D we have to project the cross product onto `u`, by using `(Cp x Ap) . u`. For dot product we don't have to change anything and simply `Cp . Ap`.

Now is we substitute `Ap = A - (A . u) u` and `Cp = C - (C . u) u` (remove the component along the `u` to get the orthogonal projection) we will get the formulas you've used here. They are just an alternative form to get the angle phi without explicitly computing `Cp` and `Ap` (by the way I think the code where `Cp` and `Ap` are computed would be also fine).